src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -879,6 +981,7 @@ private void ParseHelper(string[] args)
                 }
                 else if (MatchSwitch(switchKey, "file", "f"))
                 {
+                    ParametersUsed |= ParameterBitmap.File;
                     if (!ParseFile(args, ref i, noexitSeen))
```

Comment:
I notice some parameter info is collected regardless of whether the arg parsing will fail, like this one, and some others is collected only if the arg parsing is successful, such as `settingsfile` or `workingdirectory`. It would be nice to be consistent.